packages/@o3r/schematics/src/utility/modules.ts

```diff
@@ -213,3 +243,119 @@ export function insertBeforeModule(line: string, file: string, recorder: UpdateR
+        try {
+          require(pck.name);
+          return false;
+        } catch {
+          return true;
+        }
+      });
+  }
+
+  return packages;
+}
+
+/**
+ * Get Available Otter modules on NPMjs.org and get latest package information
+ * Similar to {@link getAvailableModules} with additional calls to retrieve all the package's information
+ *
+ * @param keyword Keyword to search for Otter modules
+ * @param scopeWhitelist List of whitelisted scopes
+ * @param onlyNotInstalled Determine if only the package that are NOT installed should be returned
+ * @param logger Logger to use to report call failure (as debug message)
+ */
+export async function getAvailableModulesWithLatestPackage(
+    keyword: string,
+    scopeWhitelist: string[] | readonly string[],
+    onlyNotInstalled = true,
+    logger?: logging.LoggerApi): Promise<NpmRegistryPackage[]> {
+
+  const packages = await getAvailableModules(keyword, scopeWhitelist, !onlyNotInstalled);
+
+  return Promise.all(
+    packages
+      .map(async (pck) => {
+        try {
+          const pckInfo = await promiseGetRequest<PackageJson>(`https://registry.npmjs.org/${pck.name}/latest`);
+          return {
+            ...pck,
+            package: pckInfo
+          };
+        } catch {
+          logger?.debug(`Failed to retrieve information for ${pck.name}`);
+          return pck;
+        }
+      })
+  );
+}
+
+/**
+ * Format a module description to be displayed in the terminal
+ *
+ * @param npmPackage Npm Package to display
+ * @param runner runner according to the context
+ * @param keywordTags Mapping of the NPM package Keywords and a displayed tag
+ * @param logger Logger to use to report package read failure (as debug message)
+ */
+export function formatModuleDescription(npmPackage: NpmRegistryPackage, runner: 'npx' | 'yarn' = 'npx', keywordTags: Record<string, string> = {}, logger?: logging.LoggerApi) {
+  let otterVersion: string | undefined;
+  const otterCorePackageName = '@o3r/core';
+  const otterCoreRange = npmPackage.package?.peerDependencies?.[otterCorePackageName];
+
+  if (npmPackage.package) {
+    try {
+      const otterCorePackage = findClosestPackageJson(require.resolve(otterCorePackageName));
+      if (otterCorePackage) {
+        const {version} = JSON.parse(fs.readFileSync(otterCorePackage, { encoding: 'utf-8'})) as PackageJson;
+        otterVersion = version;
+      }
+    } catch {
+      logger?.debug('Fail to find local Otter installation');
+    }
+  }
+
+  const flags = Object.entries(keywordTags)
+    .filter(([key]) => npmPackage.keywords?.includes(key))
+    .map(([, flag]) => flag);
+
+  const outdatedWarning = otterVersion && otterCoreRange && !satisfies(otterVersion, otterCoreRange) ? ' ' + chalk.yellow(`(outdated, supporting ${otterCoreRange})`) : '';
+
+  const lines = [
+    chalk.bold(`${runner} ng add ${chalk.cyan(npmPackage.name)}`) + outdatedWarning,
+    chalk.italic(npmPackage.description || '<no description>'),
```

Comment:
shoudn't we add a line if there is a `deprecated` message in the package json ?
(not needed for this PR)

Reply:
Are you sure that `deprecated` is officially a package.json field?